bpf/lib/trace.h

```diff
@@ -189,13 +190,26 @@ send_trace_notify(struct __ctx_buff *ctx, enum trace_point obs_point,
 	__u64 ctx_len = ctx_full_len(ctx);
 	__u64 cap_len = min_t(__u64, monitor ? : TRACE_PAYLOAD_LEN,
 			      ctx_len);
+	struct ratelimit_key rkey = {
+		.usage = RATELIMIT_USAGE_EVENTS_MAP,
+	};
+	struct ratelimit_settings settings = {
+		.topup_interval_ns = NSEC_PER_SEC,
+	};
 	struct trace_notify msg __align_stack_8;
 
 	update_trace_metrics(ctx, obs_point, reason);
 
 	if (!emit_trace_notify(obs_point, monitor))
 		return;
 
+	if (EVENTS_MAP_RATE_LIMIT_PER_SECOND > 0) {
+		settings.bucket_size = EVENTS_MAP_BURST_LIMIT;
+		settings.tokens_per_topup = EVENTS_MAP_RATE_LIMIT_PER_SECOND;
+		if (!ratelimit_check_and_take(&rkey, &settings))
```

Comment:
@cilium/sig-datapath / @dylandreimerink have we thought this through from a datapath complexity perspective, is that a concern for some of these particular event types?

Reply:
I had not. It would perhaps be a good idea to add `EVENTS_MAP_RATE_LIMIT_PER_SECOND > 0` to some of the complexity tests. 

My gut feeling is that it should not add to much, but I have not measured it, and I don't know if we are close to any limits anywhere.

Reply:
Somewhat related, would it be worthwhile to put this functionality behind a #ifdef so we can avoid the code paths if there ever are compllexity issues?

Reply:
@tommyp1ckles I think the solution to complexity issues is to ensure that we are testing the combinations on all relevant kernels and only admitting the code if we can ensure it doesn't cause problems. Adding #ifdefs means that there is yet one more combination of configuration options that needs to be tested, and we already have too many permutations to reasonably cover.

The flip side is that it's possible to micro-optimize some codepaths with #ifdefs. Ideally we don't introduce significant delays with this new functionality, but it's likely premature to implement such optimizations here.

Reply:
Seems like this thread is not actionable, so resolving it.

Reply:
Unresolving as it seems to me that @dylandreimerink's suggestion about including the rate limiter to complexity tests has been overlooked

> It would perhaps be a good idea to add `EVENTS_MAP_RATE_LIMIT_PER_SECOND > 0` to some of the complexity tests.

I'm not familiar enough with theses tests to make a call though, cc @joestringer

Reply:
If we were to add `EVENTS_MAP_RATE_LIMIT_PER_SECOND > 0` to any complexity tests, let's please do this as a follow-up to this PR.